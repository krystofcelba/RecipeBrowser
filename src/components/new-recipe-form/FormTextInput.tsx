import React from 'react';
import { StyleSheet, TextInput, StyleProp, TextStyle, Keyboard } from 'react-native';

import AppTheme from '../../assets/appTheme';

interface Props {
  placeholder?: string;
  numberOfLines?: number;
  style?: StyleProp<TextStyle>;
  input: {
    name: string;
    value: string;
    onChange: (text: string) => void;
  };
}

const FormTextInput = ({
  placeholder = '',
  numberOfLines = 1,
  style,
  input: { name, value, onChange, ...otherProps },
}: Props) => (
  <TextInput
    testID={name}
    value={typeof value === 'string' ? value : ''}
    onChangeText={onChange}
    numberOfLines={numberOfLines}
    style={[styles.input, numberOfLines > 1 && { height: numberOfLines * AppTheme.defaultFontSize }, style && style]}
    placeholder={placeholder}
    multiline={numberOfLines > 1}
    {...otherProps}
  />
);

const styles = StyleSheet.create({
  input: {
    fontFamily: AppTheme.fontFamily,
    fontSize: AppTheme.defaultFontSize,
    width: '100%',
  },
});

export default FormTextInput;
