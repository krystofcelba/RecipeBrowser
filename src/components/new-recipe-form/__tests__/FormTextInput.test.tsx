import React from 'react';
import { shallow } from 'enzyme';
import FormTextInput from '../FormTextInput';

describe('FormTextInput component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <FormTextInput placeholder="placeholder" input={{ name: '', value: '', onChange: () => {} }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with defined style', () => {
    const wrapper = shallow(
      <FormTextInput
        placeholder="placeholder"
        input={{ name: '', value: '', onChange: () => {} }}
        style={{ margin: 10 }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('call onChange as expected', () => {
    const mockedText = 'a';
    const wrapper = shallow(
      <FormTextInput
        placeholder="placeholder"
        input={{
          name: '',
          value: '',
          onChange: text => {
            expect(text).toBe(mockedText);
          },
        }}
      />,
    );
    wrapper.simulate('changeText', mockedText);
  });
});
