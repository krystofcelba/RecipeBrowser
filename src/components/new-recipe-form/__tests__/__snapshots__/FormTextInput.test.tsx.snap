// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput component renders as expected 1`] = `
<TextInput
  multiline={false}
  numberOfLines={1}
  onChangeText={[Function]}
  placeholder="placeholder"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
  testID=""
  value=""
/>
`;

exports[`FormTextInput component renders as expected multiline 1`] = `
<TextInput
  multiline={true}
  numberOfLines={3}
  onChangeText={[Function]}
  placeholder=""
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
        "width": "100%",
      },
      Object {
        "height": 42,
      },
      undefined,
    ]
  }
  testID=""
  value=""
/>
`;

exports[`FormTextInput component renders as expected with defined style 1`] = `
<TextInput
  multiline={false}
  numberOfLines={1}
  onChangeText={[Function]}
  placeholder="placeholder"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
        "width": "100%",
      },
      false,
      Object {
        "margin": 10,
      },
    ]
  }
  testID=""
  value=""
/>
`;

exports[`FormTextInput component renders as expected with non string value 1`] = `
<TextInput
  multiline={false}
  numberOfLines={1}
  onChangeText={[Function]}
  placeholder=""
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
  testID={String {}}
  value=""
/>
`;
